presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClientModule.java

```diff
@@ -68,4 +80,43 @@ private void checkConfiguration(String connectionUrl)
             throw new PrestoException(PHOENIX_CONFIG_ERROR, e);
         }
     }
+
+    @Provides
+    @Singleton
+    public ConnectionFactory getConnectionFactory(PhoenixConfig config, Properties connectionProperties)
+            throws SQLException
+    {
+        return new DriverConnectionFactory(
+                DriverManager.getDriver(config.getConnectionUrl()),
+                config.getConnectionUrl(),
+                Optional.empty(),
+                Optional.empty(),
+                connectionProperties);
+    }
+
+    @Provides
+    @Singleton
+    public Properties getConnectionProperties(PhoenixConfig config)
```

Comment:
Injecting a generic object like `Properties` globally seems like a bad idea. We should add a qualifier like `ForPhoenixClient` or remove `@Inject` from `PhoenixClient` and create a provider method for it.